grpc/jwt/jwt.go

```diff
@@ -121,16 +133,25 @@ func JWTokenSource(secretKey []byte, pem bool, clientID string) (oauth2.TokenSou
 	_ = t.Set(jwt.IssuerKey, clientID)
 	_ = t.Set(jwt.SubjectKey, clientID)
 
+	if tokenLifetime < 2*time.Minute {
+		// Don't let it be smaller than 2 min because token will be issued in every 1 minute by default.
+		tokenLifetime = 2 * time.Minute
+	} else if tokenLifetime == 0 || tokenLifetime > 24*time.Hour {
+		// No more than 24h allowed
+		tokenLifetime = time.Hour
+	}
```

Comment:
This is wrong. If tokenLifetime is not set, it will be 0, so it will be assigned to 2 minutes. The `else if` will never be reached.

I would change it a bit. I would make just 1 condition,
```
if tokenLifetime < 2*time.Minute || tokenLifetime > 24*time.Hour {
    tokenLifetime = time.Hour
}
```

This will handle all cases for you

Reply:
indeed it was completely stupid